### PR TITLE
Show UI when no PRs exists

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
**This PR depends on PR #3972 first**

When no matches exist when filtering in the PR list, it has no contents. This fixes that.

## Before
before

## After
after

_I based this PR on `show-no-prs` so the diff would be easier to read_

### CODE REVIEW

**Comment by tierninho** [comment]
:check tested

File: app/src/ui/branches/branches-container.tsx [hunk 1]
```diff
@@ -174,42 +173,37 @@ export class BranchesContainer extends React.Component<
 
   private renderPullRequests() {
     const pullRequests = this.props.pullRequests
-    if (pullRequests.length) {
+    const repo = this.props.repository
```
Comment (shiftkey) [inline app/src/ui/branches/branches-container.tsx]: :lipstick: move these closer to the component, so it's clear where they're required

File: app/src/ui/branches/branches-container.tsx [hunk 2]
```diff
@@ -174,42 +173,37 @@ export class BranchesContainer extends React.Component<
...
+      ? repo.gitHubRepository.fullName
+      : repo.name
+    const isOnDefaultBranch =
+      this.props.defaultBranch &&
+      this.props.currentBranch &&
+      this.props.defaultBranch.name === this.props.currentBranch.name
+
+    if (this.props.isLoadingPullRequests) {
+      return <PullRequestsLoading key="prs-loading" />
+    } else {
```
Comment (shiftkey) [inline app/src/ui/branches/branches-container.tsx]: :lipstick: the `else` here is unnecessary, and would reduce a bit of nesting

File: app/src/ui/branches/pull-request-list.tsx [hunk 3]
```diff
@@ -27,15 +28,23 @@ export const RowHeight = 47
 interface IPullRequestListProps {
   /** The pull requests to display. */
   readonly pullRequests: ReadonlyArray<PullRequest>
+  readonly selectedPullRequest: PullRequest | null
```
Comment (shiftkey) [inline app/src/ui/branches/pull-request-list.tsx]: 📝 docs for these new props plz

File: app/src/ui/branches/branches-container.tsx (hunk 2 above)
Comment (iAmWillShepherd) [reply app/src/ui/branches/branches-container.tsx]: True, but this is an either-or condition and not a guard clause to make sure invariants are met. I'm fine with removing the else, but I intentionally kept it around to communicate these two conditions are related.

**Comment by iAmWillShepherd** [comment]
🔢

File: app/src/ui/branches/pull-request-list.tsx [hunk 4]
```diff
@@ -28,14 +29,28 @@ interface IPullRequestListProps {
...
   readonly onItemClick: (pullRequest: PullRequest) => void
 
   /** Called when the user wants to dismiss the foldout. */
   readonly onDismiss: () => void
 
-  readonly selectedPullRequest: PullRequest | null
-
+  /** Callback to fire when the filter text is changed */
+  readonly onFilterTextChanged: (filterText: string) => void
+  readonly onCreateBranch: () => void
```
Comment (shiftkey) [inline app/src/ui/branches/pull-request-list.tsx]: 📝 docs

File: app/src/ui/branches/pull-request-list.tsx [hunk 5]
```diff
@@ -28,14 +29,28 @@ interface IPullRequestListProps {
...
+  readonly onCreatePullRequest: () => void
```
Comment (shiftkey) [inline app/src/ui/branches/pull-request-list.tsx]: 📝 docs

**Comment by iAmWillShepherd** [comment]
🔢

File: app/src/ui/branches/pull-request-list.tsx [hunk 6]
```diff
@@ -28,14 +29,34 @@ interface IPullRequestListProps {
...
+  /** Callback to fire when the filter text is changed */
+  readonly onFilterTextChanged: (filterText: string) => void
+
+  /** Called when the user opts to create a branch */
+  readonly onCreateBranch: () => void
+
+  /** Called when the user opts to create a pull request */
+  readonly onCreatePullRequest: () => void
 
+  /** Callbacked fired when user selects a new pull request */
```
Comment (shiftkey) [inline app/src/ui/branches/pull-request-list.tsx]: `s/Callbacked/callback/`

Comment (iAmWillShepherd) [reply app/src/ui/branches/pull-request-list.tsx]: Fixed in [62f9a3b](https://github.com/desktop/desktop/pull/3972/commits/62f9a3b9cfcede4c2fbcf10d8c47ed505752d99b)

**Pull request merged**